pkg/arvo/sys/hoon.hoon

```diff
@@ -5810,25 +5803,56 @@
   ::
   ++  when
     ~+
-    ;~  plug
-      %+  cook
-        |=([a=@ b=?] [b a])
-      ;~(plug dim:ag ;~(pose (cold | hep) (easy &)))
-      ;~(pfix dot mot:ag)   ::  month
-      ;~(pfix dot dip:ag)   ::  day
-      ;~  pose
-        ;~  pfix
-          ;~(plug dot dot)
-          ;~  plug
-            dum:ag
-            ;~(pfix dot dum:ag)
-            ;~(pfix dot dum:ag)
-            ;~(pose ;~(pfix ;~(plug dot dot) (most dot qix:ab)) (easy ~))
-          ==
+    |=  tub=nail
+    ^-  (like [[? @] @ @ [@ @ @ (list @ux)]])
+    =<
```

Comment:
There's a lot of non-standard indentation in this block which can makes it hard to follow at a glance

Reply:
I have improved indentation and comments. Let me know if there are more indentation/style issues.

Reply:
Just with regards to spacing & ordering, I'd recommend this:
```
++  when
  ~+
  |=  tub=nail
  ^-  (like [[? @] @ @ [@ @ @ (list @ux)]])
  =>
    |%
    ++  yelq  |=([ad=? yer=@ud] ?:(ad (yelp yer) (yelp (dec yer))))
    ++  two  (bass 10 ;~(plug sed:ab (stun [0 1] sid:ab)))
    ++  duo  (bass 10 ;~(plug sid:ab (stun [0 1] sid:ab)))
    ++  veal
      |=  [a=@ b=?]
      ?:  |(=(0 a) &(=(| b) (gth a +(jes:yo))))
        ~
      (some [b a])
    --
  ::
  =/  vex
  %.  tub
  ;~  plug
      :: year
      ::
      (sear veal ;~(plug dim:ag ;~(pose (cold | hep) (easy &))))
      :: .month
      ::
      ;~(pfix dot (sear |=(m=@ ?:((lte m 12) (some m) ~)) two))
  ==
  ::
  ?~  q.vex
    vex
  =+  yer=->:p.u.q.vex
  =+  mot=+:p.u.q.vex
  ::  day bound
  ::
  =/  mob=@
    ?:  (yelq -:p.u.q.vex)
      (snag (dec mot) moy:yo)
    (snag (dec mot) moh:yo)
  :: .day
  ::
  =/  dex
    %.  q.u.q.vex
    (sear |=(d=@ ?:((lte d mob) (some d) ~)) ;~(pfix dot two))
  ::
  ?~  q.dex
    dex
  :: ..h.m.s..f
  ::
  =/  tex
    %.  q.u.q.dex
    ;~  pose
        ;~  pfix
            ;~(plug dot dot)
            ;~  plug
                (sear |=(h=@ ?:((lte h 23) (some h) ~)) duo)
                ;~(pfix dot (sear |=(m=@ ?:((lte m 59) (some m) ~)) duo))
                ;~(pfix dot (sear |=(s=@ ?:((lte s 59) (some s) ~)) duo))
                ;~(pose ;~(pfix ;~(plug dot dot) (most dot qix:ab)) (easy ~))
        ==  ==
        (easy [0 0 0 ~])
    ==
  ::
  ?~  q.tex
    tex
  :: [year month day time]
  ::
  [p.tex [~ [[-:p.u.q.vex +:p.u.q.vex p.u.q.dex p.u.q.tex] q.u.q.tex]]]
```

You have some really wide expressions that I would usually make taller in *my own* code, but they're still readable and I think the way you have them is on-brand for how the rest of `hoon.hoon` is written.